arch/arm/src/s32k3xx/hardware/s32k3xx_qspi.h

```diff
@@ -0,0 +1,505 @@
+#define QSPI_LUT_OPRND0(n)                ((n << QSPI_LUT_OPRND0_SHIFT) & QSPI_LUT_OPRND0_MASK)
+#define QSPI_LUT_PAD0_SHIFT               (8)       /* Bits 8-9: Pad information for INSTR0 (PAD0) */
+#define QSPI_LUT_PAD0_MASK                (0x03 << QSPI_LUT_PAD0_SHIFT)
+#  define QSPI_LUT_PAD0_1                 (0x00 << QSPI_LUT_PAD0_SHIFT) /* 1 Pad */ 
+#  define QSPI_LUT_PAD0_2                 (0x01 << QSPI_LUT_PAD0_SHIFT) /* 2 Pad */ 
+#  define QSPI_LUT_PAD0_4                 (0x02 << QSPI_LUT_PAD0_SHIFT) /* 4 Pad */ 
+
+#define QSPI_LUT_INSTR0_SHIFT             (10)      /* Bits 10-15: Instruction 0 (INSTR0) */
+#define QSPI_LUT_INSTR0_MASK              (0x3f << QSPI_LUT_INSTR0_SHIFT)
+#define QSPI_LUT_INSTR0(n)                ((n << QSPI_LUT_INSTR0_SHIFT) & QSPI_LUT_INSTR0_MASK)
+
+#define QSPI_LUT_OPRND1_SHIFT             (16)       /* Bits 16-23: Operand for INSTR1 (OPRND1) */
+#define QSPI_LUT_OPRND1_MASK              (0xff << QSPI_LUT_OPRND1_SHIFT)
+#define QSPI_LUT_OPRND1(n)                ((n << QSPI_LUT_OPRND1_SHIFT) & QSPI_LUT_OPRND1_MASK)
+#define QSPI_LUT_PAD1_SHIFT               (24)       /* Bits 24-25: Pad information for INSTR1 (PAD1) */
+#define QSPI_LUT_PAD1_MASK                (0x03 << QSPI_LUT_PAD1_SHIFT)
+#  define QSPI_LUT_PAD1_1                 (0x00 << QSPI_LUT_PAD1_SHIFT) /* 1 Pad */ 
+#  define QSPI_LUT_PAD1_2                 (0x01 << QSPI_LUT_PAD1_SHIFT) /* 2 Pad */ 
+#  define QSPI_LUT_PAD1_4                 (0x02 << QSPI_LUT_PAD1_SHIFT) /* 4 Pad */ 
+
+#define QSPI_LUT_INSTR1_SHIFT             (26)      /* Bits 26-31: Instruction 1 (INSTR1) */
+#define QSPI_LUT_INSTR1_MASK              (0x3f << QSPI_LUT_INSTR1_SHIFT)
+#define QSPI_LUT_INSTR1(n)                ((n << QSPI_LUT_INSTR1_SHIFT) & QSPI_LUT_INSTR1_MASK)
+
+/* External Memory Base Address */
+
+#define QSPI_AMBA_BASE    0x68000000
+
+/* flash connection to the QSPI module */
+
+typedef enum
+{
+    QSPI_SIDE_A1    = 0x00u,  /* Serial flash connected on side A1    */
+    QSPI_SIDE_A2    = 0x01u,  /* Serial flash connected on side A2    */
+    QSPI_SIDE_B1    = 0x02u,  /* Serial flash connected on side B1    */
+    QSPI_SIDE_B2    = 0x03u,  /* Serial flash connected on side B2    */
+} s32k3xx_qspi_connectiontype;
+
+/* flash operation type */
+
+typedef enum
+{
+    QSPI_OP_TYPE_CMD          = 0x00u,  /* Simple command                              */
+    QSPI_OP_TYPE_WRITE_REG    = 0x01u,  /* Write value in external flash register      */
+    QSPI_OP_TYPE_RMW_REG      = 0x02u,  /* RMW command on external flash register      */
+    QSPI_OP_TYPE_READ_REG     = 0x03u,  /* Read external flash register until expected value is read    */
+    QSPI_OP_TYPE_QSPI_CFG     = 0x04u,  /* Re-configure QSPI controller                */
+} s32k3xx_qspi_optype;
+
+/* Lut commands */
+
+typedef enum
+{
+    QSPI_LUT_INSTR_STOP            = (0U << 10U),    /* End of sequence                           */
```

Comment:
please use 2 spaces for indentation and either `u` or `U` across a single file, but not mix